net/adoptopenjdk/bumblebench/string/StringIndexOfStringBench.java

```diff
@@ -0,0 +1,171 @@
+	static boolean verifyResult = option("verify", 0) != 0;
+
+	static String[] strings = new String[MAX_NUM_STRINGS];
+	static int[] indices = new int[MAX_NUM_STRINGS];
+	static int[] resultIndices = new int[MAX_NUM_STRINGS];
+
+	static int currentIterationCount;
+	static int minIndex, maxIndex;
+	static int minLen, maxLen;
+
+	private static String longString_64K_char_compressible;
+
+	static {
+		minIndex = 1;
+		maxIndex = 50;
+		// very short strings (e.g. single-char-strings) at high index aren't worth
+		// testing because they are not unique at all
+		// and can appear at low index, in which case they are the same as
+		// single-char-string at low index.
+		// start with 8-char so that the sub-strings are more unique.
+		minLen = 8;
+		maxLen = 20;
+
+		switch (stringLength) {
+		case 1:
+			// short sub-string (1-20)
+			// use default numbers
+			break;
+		case 2:
+			// medium string length, around 100
+			minLen = 80;
+			maxLen = 120;
+			break;
+		case 3:
+			// long string, around 1k
+			minLen = 800;
+			maxLen = 1200;
+			break;
+		case 4:
+			// ultra long string, around 10k
+			minLen = 9000;
+			maxLen = 11000;
+			break;
+		default:
+			break;
+		}
+
+		// the whole string is 63k char long
+		switch (indexLevel) {
+		case 1:
+			// low index
+			break;
+		case 2:
+			// medium index, around 100
+			minIndex = 80;
+			maxIndex = 120;
+			break;
+		case 3:
+			// high index, around 1k
+			minIndex = 800;
+			maxIndex = 1200;
+			break;
+		case 4:
+			// ultra high index, around 50k
+			minIndex = 48000;
+			maxIndex = 51000;
+			break;
+		default:
+			break;
+		}
+
+		// fixed random sequence
+		Random rand = new Random();
+		rand.setSeed(68798243246987L);
+
+		// generate a random master string
+		int stringLen = 64000;
+		StringBuffer sb = new StringBuffer();
+		sb.setLength(stringLen);
+
+		for (int i = 0; i < stringLen; ++i) {
+			int randInt = rand.nextInt(256);
+			sb.setCharAt(i, (char) randInt);
+		}
+
+		longString_64K_char_compressible = sb.toString();
+
+		// Pre-generate an array of sub-strings
+		for (int i = 0; i < MAX_NUM_STRINGS; i++) {
+			int randLen = rand.nextInt(maxLen - minLen + 1) + minLen;
+			int randIndex = rand.nextInt(maxIndex - minIndex + 1) + minIndex;
+			strings[i] = longString_64K_char_compressible.substring(randIndex, randIndex + randLen);
+			indices[i] = randIndex;
+		}
+	}
+
+	protected int maxIterationsPerLoop() {
+		return MAX_NUM_STRINGS;
+	}
+
+	int _escape; // To prevent the optimizer from removing the whole method
+
+	int doIndexOfString() {
+
+		for (int i = 0; i < MAX_NUM_STRINGS; i++) {
+			_escape = longString_64K_char_compressible.indexOf(strings[i]);
+			resultIndices[i] = _escape;
+		}
+
+		return MAX_NUM_STRINGS;
+	}
+
+	protected long doBatch(long numLoops) throws InterruptedException {
+
+		for (long loop = 0; loop < numLoops; loop++) {
+			startTimer();
+			int total = doIndexOfString();
+			pauseTimer();
+			_escape = total;
+
+			// verify
+			if (verifyResult) {

```

Comment:
I suppose it doesn't hurt, but this is a benchmark. We should be assuming the results of the operations are correct. Functional testing should take care that the `indexOf` operation gives correct results. I'm not sure this code belongs in a performance benchmark.